### ISSUE TITLE
Action and Selector decorators' parameters

### ISSUE THREAD

**Issue by zygimantas** [ngxs/store#369]
### Versions

```
* ngxs: 3.0.1
* @angular/core: 6.0.0
* typescript 2.7.2
```

### Repro steps

* Step 1 use Action or Selector decorator
* Step 2 convert the function into expression

### Observed behavior

```
Error: [ts] Unable to resolve signature of property decorator when called as an expression.
```

### Desired behavior

No error.

### Mention any other details that might be useful (optional)

https://github.com/Microsoft/TypeScript/issues/19528

**Comment by amcdnl** [comment]
Can you provide a code sample?

**Comment by zygimantas** [comment]
https://stackblitz.com/edit/angular-fq6yzk

**Comment by zygimantas** [comment]
check ```app/app.state.ts``` file

**Comment by kyusupov33** [comment]
@markwhitfeld what with this issue?

**Comment by splincode** [comment]
fixed in 3.4.0
